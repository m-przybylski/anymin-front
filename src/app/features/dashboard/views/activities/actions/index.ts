import * as ActivitiesActions from './activities.actions';
import * as ActivitiesApiActions from './activities-api.actions';
import * as ActivitiesPageActions from './activities-page.actions';
import * as BalanceApiActions from './balance-api.actions';
import * as ActivitiesWsActions from './activities-ws.actions';

export { ActivitiesActions, ActivitiesApiActions, ActivitiesPageActions, BalanceApiActions, ActivitiesWsActions };
