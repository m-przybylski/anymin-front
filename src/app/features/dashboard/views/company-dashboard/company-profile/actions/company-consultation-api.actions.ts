// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { ICompanyConsultationDetails, ICompanyEmployeeRowComponent } from '../services/company-consultation.service';

export enum CompanyConsultationApiActionTypes {
  LoadSuccess = '[Comp consult API] Load consultation success',
  LoadFailure = '[Comp consult API] Load consultation failure',
  LoadInvitesSuccess = '[Comp consult API] Load invites success',
  LoadInvitesFailure = '[Comp consult API] Load invites failure',
  DeleteEmploymentSuccess = '[Comp consult API] Delete employment success',
  DeleteEmploymentFailure = '[Comp consult API] Delete employment failure',
  DeleteInviteSuccess = '[Comp consult API] Delete invite success',
  DeleteInviteFailure = '[Comp consult API] Delete invite failure',
}

export class LoadConsultationSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadSuccess;

  constructor(public payload: ICompanyConsultationDetails) {}
}

export class LoadConsultationFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadFailure;

  constructor(public payload: any) {}
}

export class LoadInvitesSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadInvitesSuccess;

  constructor(public payload: ReadonlyArray<ICompanyEmployeeRowComponent>) {}
}

export class LoadInvitesFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.LoadInvitesFailure;

  constructor(public payload: any) {}
}

export class DeleteEmploymentSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteEmploymentSuccess;

  constructor(public payload: string) {}
}

export class DeleteEmploymentFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteEmploymentFailure;

  constructor(public payload: any) {}
}

export class DeleteInviteSuccessAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteInviteSuccess;

  constructor(public payload: string) {}
}

export class DeleteInviteFailureAction implements Action {
  public readonly type = CompanyConsultationApiActionTypes.DeleteInviteFailure;

  constructor(public payload: any) {}
}

export type CompanyConsultationApiActionsUnion =
  | LoadConsultationSuccessAction
  | LoadConsultationFailureAction
  | LoadInvitesSuccessAction
  | LoadInvitesFailureAction
  | DeleteEmploymentSuccessAction
  | DeleteEmploymentFailureAction;
