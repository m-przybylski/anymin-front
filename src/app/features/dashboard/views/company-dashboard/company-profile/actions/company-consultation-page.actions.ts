// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { GetSessionWithAccount } from '@anymind-ng/api';
import { UserTypeEnum } from '@platform/core/reducers/navbar.reducer';

export enum CompanyConsultationPageActionTypes {
  Load = '[Comp consult page] Load consultation',
  LoadInvites = '[Comp consult page] Load pending invites',
  Clear = '[Comp consult page] Clear',
  DeleteEmployment = '[Comp consult page] Delete employment',
  DeleteInvite = '[Comp consult page] Delete invite',
}

export class LoadConsultationsAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.Load;

  constructor(
    public payload: {
      serviceId: string;
      getSessionWithAccount?: GetSessionWithAccount;
      userType?: UserTypeEnum;
    },
  ) {}
}

export class LoadPendingInvitesAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.LoadInvites;

  constructor(public payload: string) {}
}

export class ClearAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.Clear;
}

export class DeleteEmploymentAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.DeleteEmployment;

  constructor(public payload: string) {}
}

export class DeleteInviteAction implements Action {
  public readonly type = CompanyConsultationPageActionTypes.DeleteInvite;

  constructor(public payload: string) {}
}

export type CompanyConsultationPageActionsUnion =
  | LoadConsultationsAction
  | LoadPendingInvitesAction
  | ClearAction
  | DeleteEmploymentAction
  | DeleteInviteAction;
