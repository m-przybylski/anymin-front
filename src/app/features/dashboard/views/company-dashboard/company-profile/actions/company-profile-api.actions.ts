// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';
import { ServiceWithEmployments } from '@anymind-ng/api';

export enum CompanyProfileApiActionTypes {
  LoadCompanyConsultationsSuccess = '[Comp profile API] Load consultations Success',
  LoadCompanyConsultationsFailure = '[Comp profile API] Load consultations Failure',
  DeleteEmploymentSuccess = '[Comp profile API] Delete employment Success',
  DeleteEmploymentFailure = '[Comp profile API] Delete employment Failure',
}

export class LoadCompanyConsultationsSuccessAction implements Action {
  public readonly type = CompanyProfileApiActionTypes.LoadCompanyConsultationsSuccess;
  constructor(public payload: ReadonlyArray<ServiceWithEmployments>) {}
}

export class DeleteEmploymentSuccessAction implements Action {
  public readonly type = CompanyProfileApiActionTypes.DeleteEmploymentSuccess;
  constructor(public payload: string) {}
}
export type CompanyProfileApiActionsUnion = LoadCompanyConsultationsSuccessAction | DeleteEmploymentSuccessAction;
