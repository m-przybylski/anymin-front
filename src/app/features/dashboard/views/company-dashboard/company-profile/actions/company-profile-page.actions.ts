// tslint:disable:max-classes-per-file
import { Action } from '@ngrx/store';

export enum CompanyProfilePageActionTypes {
  LeaveConsultation = '[Comp profile page] Leave consultation',
  RemoveExpertFromConsultation = '[Comp profile page] Remove expert from consultation',
}

export class LeaveConsultationAction implements Action {
  public readonly type = CompanyProfilePageActionTypes.LeaveConsultation;
  constructor(public payload: string) {}
}

export class RemoveExpertFromConsultationAction implements Action {
  public readonly type = CompanyProfilePageActionTypes.RemoveExpertFromConsultation;
  constructor(public payload: string) {}
}

export type CompanyProfilePageActionsUnion = LeaveConsultationAction | RemoveExpertFromConsultationAction;
