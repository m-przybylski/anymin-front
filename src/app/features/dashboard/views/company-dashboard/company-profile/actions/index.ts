import * as CompanyProfileApiActions from './company-profile-api.actions';
import * as CompanyProfilePageActions from './company-profile-page.actions';
import * as CompanyConsultationPageActions from './company-consultation-page.actions';
import * as CompanyConsultationApiActions from './company-consultation-api.actions';

export {
  CompanyProfileApiActions,
  CompanyProfilePageActions,
  CompanyConsultationPageActions,
  CompanyConsultationApiActions,
};
