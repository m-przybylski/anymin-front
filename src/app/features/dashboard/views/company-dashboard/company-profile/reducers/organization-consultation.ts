import { ICompanyConsultationDetails, ICompanyEmployeeRowComponent } from '../services/company-consultation.service';
import { CompanyConsultationPageActions, CompanyConsultationApiActions } from '../actions';

export interface IState {
  isLoading: boolean;
  isInviteLoading: boolean;
  organizationConsultation?: ICompanyConsultationDetails;
  pendingInvitations?: ReadonlyArray<ICompanyEmployeeRowComponent>;
  error?: any;
}

const initialState: IState = {
  isLoading: false,
  isInviteLoading: false,
};

type ActionUnion =
  | CompanyConsultationPageActions.CompanyConsultationPageActionsUnion
  | CompanyConsultationApiActions.CompanyConsultationApiActionsUnion;

// tslint:disable-next-line: cyclomatic-complexity only-arrow-functions
export function reducer(state: IState = initialState, action: ActionUnion): IState {
  switch (action.type) {
    case CompanyConsultationPageActions.CompanyConsultationPageActionTypes.Load:
      return {
        ...state,
        isLoading: true,
        isInviteLoading: true,
      };
    case CompanyConsultationPageActions.CompanyConsultationPageActionTypes.LoadInvites:
      return {
        ...state,
        isInviteLoading: true,
      };
    case CompanyConsultationPageActions.CompanyConsultationPageActionTypes.Clear:
      return {
        ...initialState,
      };
    case CompanyConsultationApiActions.CompanyConsultationApiActionTypes.DeleteEmploymentSuccess:
      return {
        ...state,
        organizationConsultation: state.organizationConsultation && {
          ...state.organizationConsultation,
          employeesList: state.organizationConsultation.employeesList.filter(
            employee => employee.id !== action.payload,
          ),
        },
      };
    case CompanyConsultationApiActions.CompanyConsultationApiActionTypes.LoadSuccess:
      return {
        ...state,
        isLoading: false,
        organizationConsultation: action.payload,
      };
    case CompanyConsultationApiActions.CompanyConsultationApiActionTypes.LoadFailure:
      return {
        ...state,
        isLoading: false,
        error: action.payload,
      };
    case CompanyConsultationApiActions.CompanyConsultationApiActionTypes.LoadInvitesSuccess:
      return {
        ...state,
        isInviteLoading: false,
        pendingInvitations: action.payload,
      };
    case CompanyConsultationPageActions.CompanyConsultationPageActionTypes.DeleteInvite:
      return {
        ...state,
        pendingInvitations:
          state.pendingInvitations && state.pendingInvitations.filter(invite => invite.id !== action.payload),
      };
    default:
      return state;
  }
}

export const getData = (state: IState): ICompanyConsultationDetails | undefined => state.organizationConsultation;
export const getInvites = (state: IState): ReadonlyArray<ICompanyEmployeeRowComponent> | undefined =>
  state.pendingInvitations;
export const getIsLoading = (state: IState): boolean => state.isLoading;
export const getIsInviteLoading = (state: IState): boolean => state.isInviteLoading;
