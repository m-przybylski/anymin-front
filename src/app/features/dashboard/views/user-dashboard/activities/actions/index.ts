import * as ExpertActivitiesActions from './expert-activities.actions';
import * as ExpertActivitiesApiActions from './expert-activities-api.actions';
import * as ExpertActivitiesPageActions from './expert-activities-page.actions';
import * as BalanceApiActions from './balance-api.actions';
import * as ExpertActivitiesWsActions from './expert-activities-ws.actions';

export {
  ExpertActivitiesActions,
  ExpertActivitiesApiActions,
  ExpertActivitiesPageActions,
  BalanceApiActions,
  ExpertActivitiesWsActions,
};
