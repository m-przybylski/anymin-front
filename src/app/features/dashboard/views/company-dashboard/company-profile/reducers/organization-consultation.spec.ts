import { reducer, IState } from './organization-consultation';
import { CompanyConsultationPageActions, CompanyConsultationApiActions } from '../actions';
import { DeepPartial } from 'testing/testing';

const initialState: IState = {
  isLoading: false,
  isInviteLoading: false,
};

describe('company consultation reducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const result = reducer(undefined, {} as any);

      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadConsultationsAction', () => {
    it('should start loading consultations', () => {
      const result = reducer(
        initialState,
        new CompanyConsultationPageActions.LoadConsultationsAction({
          serviceId: 'fake serviceId',
          getSessionWithAccount: undefined,
        }),
      );
      expect(result).toMatchSnapshot();
    });
  });

  describe('LoadConsultationSuccessAction', () => {
    it('should update state', () => {
      const state = {
        isLoading: true,
        isInviteLoading: false,
      };
      const result = reducer(
        state,
        new CompanyConsultationApiActions.LoadConsultationSuccessAction('fake payload' as any),
      );
      expect(result).toMatchSnapshot();
    });
  });

  describe('ClearAction', () => {
    it('should update state', () => {
      const state: any = {
        isLoading: false,
        isInviteLoading: false,
        organizationConsultation: 'fake consultation',
        pendingInvitations: [],
      };
      const result = reducer(state, new CompanyConsultationPageActions.ClearAction());
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadConsultationFailureAction', () => {
    it('should update state', () => {
      const state = {
        isLoading: true,
        isInviteLoading: false,
      };
      const result = reducer(
        state,
        new CompanyConsultationApiActions.LoadConsultationFailureAction('fake error' as any),
      );
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadInvitesAction', () => {
    it('should update state', () => {
      const state = {
        isLoading: false,
        isInviteLoading: false,
      };
      const result = reducer(state, new CompanyConsultationPageActions.LoadPendingInvitesAction('fake service'));
      expect(result).toMatchSnapshot();
    });
  });
  describe('LoadInvitesSuccessAction', () => {
    it('should update state', () => {
      const state = {
        isLoading: true,
        isInviteLoading: false,
      };
      const result = reducer(state, new CompanyConsultationApiActions.LoadInvitesSuccessAction('fake invites' as any));

      expect(result).toMatchSnapshot();
    });
  });

  describe('DeleteEmployment', () => {
    it('should update state', () => {
      const state: DeepPartial<IState> = {
        isLoading: true,
        isInviteLoading: false,
        organizationConsultation: {
          employeesList: [
            {
              id: '1234',
            },
            {
              id: '2345',
            },
            {
              id: '3456',
            },
          ],
        },
      };
      const result = reducer(
        state as any,
        new CompanyConsultationApiActions.DeleteEmploymentSuccessAction('1234' as any),
      );

      expect(result.organizationConsultation && result.organizationConsultation.employeesList.length).toBe(2);
    });
  });
});
