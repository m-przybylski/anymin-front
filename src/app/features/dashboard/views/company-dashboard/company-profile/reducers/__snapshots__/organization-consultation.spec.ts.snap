// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company consultation reducer ClearAction should update state 1`] = `
Object {
  "isInviteLoading": false,
  "isLoading": false,
}
`;

exports[`company consultation reducer LoadConsultationFailureAction should update state 1`] = `
Object {
  "error": "fake error",
  "isInviteLoading": false,
  "isLoading": false,
}
`;

exports[`company consultation reducer LoadConsultationSuccessAction should update state 1`] = `
Object {
  "isInviteLoading": false,
  "isLoading": false,
  "organizationConsultation": "fake payload",
}
`;

exports[`company consultation reducer LoadConsultationsAction should start loading consultations 1`] = `
Object {
  "isInviteLoading": true,
  "isLoading": true,
}
`;

exports[`company consultation reducer LoadInvitesAction should update state 1`] = `
Object {
  "isInviteLoading": true,
  "isLoading": false,
}
`;

exports[`company consultation reducer LoadInvitesSuccessAction should update state 1`] = `
Object {
  "isInviteLoading": false,
  "isLoading": true,
  "pendingInvitations": "fake invites",
}
`;

exports[`company consultation reducer undefined action should return the default state 1`] = `
Object {
  "isInviteLoading": false,
  "isLoading": false,
}
`;
